src/transformers/quantizers/quantizer_bnb.py

```diff
@@ -0,0 +1,449 @@
+# Copyright 2024 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import importlib
+from typing import TYPE_CHECKING, Any, Dict, List, Tuple, Union
+
+from packaging import version
+
+from .base import HFQuantizer
+
+
+if TYPE_CHECKING:
+    from ..modeling_utils import PreTrainedModel
+
+from ..utils import is_accelerate_available, is_bitsandbytes_available, is_torch_available, logging
+
+
+if is_torch_available():
+    import torch
+
+if is_bitsandbytes_available():
+    import bitsandbytes as bnb
+
+    from ..pytorch_utils import Conv1D
```

Comment:
we import both torch and bnb here if we import anything from here. -> slow?

Reply:
Indeed I think the bnb import can be done in the local level, I will modify that

Reply:
Done !